data-access-layer/iaas/AliClient.js

```diff
@@ -139,15 +147,152 @@ class AliClient extends BaseCloudClient {
       .catchThrow(SyntaxError, new UnprocessableEntity(`Object '${file}' data unprocessable`));
   }
 
-  createDiskFromSnapshot(snapshotId, zone, opts = {}) {}
+  _getComputeRequestParams(params, tags) {
+    let reqParams = _.assign({}, params);
+    let i = 1;
+    _.forEach(tags, (value, key) => {
+      let tag = {};
+      tag['Tag.' + i + '.Key'] = key;
+      tag['Tag.' + i + '.Value'] = value;
+      _.extend(reqParams, tag);
+    });
+    return reqParams;
+  }
 
-  getDiskMetadata(diskCid, zone) {}
+  createDiskFromSnapshot(snapshotId, zone, opts = {}) {
+    const params = {
+      'RegionId': this.settings.region,
+      'ZoneId': zone,
+      'SnapshotId': snapshotId,
+      'DiskCategory': opts.type || 'cloud_ssd'
+    };
+    const tags = _.assign({}, opts.tags || {}, {
+      createdBy: 'service-fabrik'
+    });
+    const reqParams = this._getComputeRequestParams(params, tags);
+    const requestOption = {
+      timeout: CONST.ALI_CLIENT.ECS.REQ_TIMEOUT,
+      method: 'POST'
+    };
+    return Promise
+      .try(() => {
+        return this.computeClient
+          .request('CreateDisk', reqParams, requestOption)
+          .tap(result => logger.info(`Created disk ${result.DiskId} from snapshot ${snapshotId}, now wait for it to be available...`))
+          .then(result => this._waitForDiskAvailability(result.DiskId))
+          .tap(diskDetails => logger.info(`Created disk ${diskDetails.DiskId} from snapshot ${snapshotId} is now Available with status: ${diskDetails.Status}`))
```

Comment:
Can you return `diskDetails` in a format similar to output of `getDiskMetadata `. For example, consider `AzureClient`: https://github.com/cloudfoundry-incubator/service-fabrik-broker/blob/rel-2019.T03b/data-access-layer/iaas/AzureClient.js#L68.

Currently BoshBasedRestore Operator doesn't have a tight dependency on this diskInfo format. However we store information of old and new disks created as part of restore operation and it would be nice to have formats for both of them to be same.

Reply:
👍

Reply:
Done